### PR TITLE
Add MlflowLoggerHook

### PR DESCRIPTION
**Pull request by daavoo**
This P.R. adds a new class to `mmcv.runner.hooks.logger` for logging metrics and (optionally) a trained model to [MLflow](https://www.mlflow.org/docs/latest/index.html); similar to existing options for pavi, wandb or tensorboard.

An unit test following the existing ones is included.

The `wandb` unit test has been moved from `test_runner `to `test_hooks`.

### CODE REVIEW

File: tests/test_hooks.py [hunk 1]
```diff
@@ -3,17 +3,24 @@
...
+try:
+    import torch
+    import torch.nn as nn
+    from torch.utils.data import DataLoader
+except ImportError:
+    warnings.warn('torch is not available')
+    torch = None
+
+only_if_torch_available = pytest.mark.skipif(
+    torch is None, reason='torch is not available')
```
Comment (hellock) [inline tests/test_hooks.py:19]: PyTorch will be made a compulsive requirement.

File: mmcv/runner/hooks/logger/mlflow.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+from mmcv.runner import master_only
+from ..hook import HOOKS
+from .base import LoggerHook
+
+
+@HOOKS.register_module
+class MlflowLoggerHook(LoggerHook):
+
+    def __init__(self,
+                 experiment_name=None,
```
Comment (hellock) [inline mmcv/runner/hooks/logger/mlflow.py:13]: Maybe simply `exp_name`?

Comment (daavoo) [reply mmcv/runner/hooks/logger/mlflow.py:13]: Done!

File: tests/test_hooks.py (hunk 1 above)
Comment (daavoo) [reply tests/test_hooks.py]: Decorators have been removed

**Pull request merged**